feature/google/src/main/java/com/revenuecat/purchases/google/findDefaultOffer.kt

```diff
@@ -35,25 +36,18 @@ private val GoogleSubscriptionOption.freePricingPhase: PricingPhase?
 private val GoogleSubscriptionOption.nonFreePricingPhase: PricingPhase?
     get() = pricingPhases.firstOrNull()?.takeIf { it.price.amountMicros > 0L }
 
-private const val DAYS_IN_YEAR = 365
-private const val DAYS_IN_MONTH = 30
+private const val DAYS_IN_DAY = 1
 private const val DAYS_IN_WEEK = 7
-
-// Would use Duration.parse but only available API 26 and up
-internal fun parseBillPeriodToDays(period: String): Int {
-    // Takes from https://stackoverflow.com/a/32045167
-    val regex = "^P(?!\$)(\\d+(?:\\.\\d+)?Y)?(\\d+(?:\\.\\d+)?M)?(\\d+(?:\\.\\d+)?W)?(\\d+(?:\\.\\d+)?D)?\$"
-        .toRegex()
-        .matchEntire(period)
-
-    regex?.let { periodResult ->
-        val toInt = fun(part: String): Int {
-            return part.dropLast(1).toIntOrNull() ?: 0
-        }
-
-        val (year, month, week, day) = periodResult.destructured
-        return (toInt(year) * DAYS_IN_YEAR) + (toInt(month) * DAYS_IN_MONTH) + (toInt(week) * DAYS_IN_WEEK) + toInt(day)
-    }
-
-    return 0
+private const val DAYS_IN_MONTH = 30
+private const val DAYS_IN_YEAR = 365
```

Comment:
Hmm would this cause issues for leap years? Or months with not 30 days? I guess it’s only used internally to compare offers, so it’s probably ok.

Reply:
Yeah, its not the most accurate date logic but its really only used to find the longest free trial so everything will be inaccurate together 🤷‍♂️